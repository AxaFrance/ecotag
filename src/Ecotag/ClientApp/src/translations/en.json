{
  "header": {
    "title": "An eco-system that aims to be eco-friendly",
    "errors": {
      "not_connected": "Not connected"
    }
  },
  "paging": {
    "display": "Display",
    "elements": "elements",
    "previous": "« Previous",
    "next": "Next »"
  },
  "stepper": {
    "confirm": "Confirm"
  },
  "resilience": {
    "error": {
      "first_part": "An error occurred.",
      "second_part": "We are aware of the problem. An investigation is ongoing..."
    },
    "forbidden": "You are not allowed to access this resource."
  },
  "not_found": {
    "title": "404 Page Not Found",
    "not": "not",
    "found": "found",
    "message": "The page you are looking for is not here !"
  },
  "home": {
    "title": "Home",
    "projects": "Projects",
    "datasets": "Datasets",
    "groups": "Teams",
    "errors": {
      "no_roles": "No role is associated to your profile."
    }
  },
  "group": {
    "title": "Teams",
    "subtitle": "Annotation teams governance",
    "creation_title": "Creation",
    "new": {
      "name_input": {
        "label": "Name : ",
        "help_message": "Ex : bank account details annotators"
      },
      "validation": "Validate"
    },
    "count_teams": {
      "first_part": "There is/are (",
      "second_part": ") team(s)"
    },
    "empty_list": "No element",
    "list": {
      "header": {
        "name": "Name",
        "users": "Users",
        "actions": "Actions"
      },
      "edit": "Edit",
      "edit_popup": {
        "header": "Add/Remove users from this team",
        "add_user_placeholder": "Add user",
        "footer": {
          "cancel": "Cancel",
          "validation": "Validate"
        }
      }
    }
  },
  "dataset": {
    "title": "Datasets",
    "subtitle": "A dataset is the representation of a collection of data",
    "new_dataset": "New dataset",
    "access_token": "Access token",
    "active_datasets_counter": "Your active datasets : ",
    "filter_by": "Filter by",
    "name": "Name",
    "empty_list": "No elements",
    "list": {
      "actions_titles": {
        "locked": "locked",
        "locked_and_work_in_progress": "locked and work in progress",
        "pending": "pending",
        "none": "unlocked"
      },
      "header": {
        "lock": "Lock",
        "name": "Name",
        "team": "Team",
        "classification": "Classification",
        "nb_files": "Number of files",
        "type": "Type",
        "creation_date": "Creation date",
        "action": "Action"
      },
      "edit": "Edit"
    },
    "new": {
      "title": "New dataset",
      "stepper": {
        "title": "New dataset"
      },
      "form_title": "New dataset",
      "name_input": {
        "label": "Name",
        "help_message": "Ex : bank-account-details-dataset"
      },
      "team_input": {
        "label": "Team"
      },
      "type_input": {
        "label": "Type",
        "help_message": "Ex : Image, Text",
        "options_labels": {
          "image": "Images (.jpg, .png, jpeg)",
          "document": "Documents (.txt, .eml, .tiff, .pdf, .doc, etc.)",
          "text": "Text (.txt)",
          "eml": "Emails (.eml)"
        },
        "help": {
          "title": "Information on the accepted documents types",
          "images": {
            "name": "Images",
            "types": " : .jpg, .png, jpeg"
          },
          "documents": {
            "name": "Documents",
            "types": " : .txt, .eml, .msg, .jpg, .png, jpeg, .tiff, .tif, .pdf, .doc, .docx, .xls, .xlsx, .ppt, .pptx, .rtf, .odt, .ods, .odp, .zip"
          },
          "text": {
            "name": "Text",
            "types": " : .txt"
          },
          "eml": {
            "name": "Eml",
            "types": " : .eml"
          }
        }
      },
      "classification_input": {
        "label": "Classification",
        "help_message": "Ex : Critical",
        "options": {
          "public": {
            "label": "Public"
          },
          "internal": {
            "label": "Internal"
          },
          "confidential": {
            "label": "Confidential"
          },
          "critical": {
            "label": "Critical"
          }
        },
        "help": {
          "title": "Information on criticality",
          "public": {
            "name": "Public",
            "detail": " : a piece of information made deliberately for the public, or specifically made for public diffusion"
          },
          "internal": {
            "name": "Internal",
            "detail": " : a piece of information that is accessible to all or a large portion of employees and is not intended for external parties."
          },
          "confidential": {
            "name": "Confidential",
            "detail": " : a piece of information restricted to a specific audience"
          },
          "critical": {
            "name": "Critical",
            "detail": " : Please refer to your manager."
          }
        }
      },
      "import_input": {
        "label": "Import from a blob : "
      },
      "dataset_to_import": {
        "label": "Dataset to import : "
      },
      "validate": "Validation"
    },
    "confirm": {
      "title": "Confirmation",
      "stepper": "New dataset",
      "confirmation_title": "New dataset added !",
      "confirmation_description": "If you have imported a dataset, the process of adding files can take a while.",
      "return_button_label": "Go back to datasets list"
    },
    "edit": {
      "confirm_modal": {
        "title": "Do you want to lock the dataset permanently ?",
        "warning_first_part": "This action is permanent.",
        "warning_second_part": "Every modification (addition or deletion of files) will be impossible then."
      },
      "subtitle": "Dataset edition",
      "lock": "Lock",
      "locked_state": {
        "locked": "Dataset locked",
        "locked_and_work_in_progress": "Dataset is being worked on",
        "default": "Dataset is being imported"
      },
      "file_upload": {
        "title": "Upload files ",
        "file_input": {
          "label": "Browse",
          "placeholder": "Drag/drop your files here"
        },
        "send_button": {
          "label": "Send"
        },
        "errors": {
          "already_existing": {
            "first_part": "The file ",
            "second_part": " already exists"
          },
          "could_not_upload": {
            "first_part": "The file ",
            "second_part": " could not be uploaded"
          }
        }
      },
      "files_list": {
        "global_info_tab": {
          "title": "Global information",
          "name": "Name",
          "creation_date": "Creation date",
          "type": "Type",
          "nb_files": "Number of files",
          "files_total_size": "Total files size",
          "files_mean_size": "Average file size"
        },
        "files_list_tab": {
          "title": "Files list",
          "name": "Name",
          "type": "Type",
          "size": "Size",
          "action": "Action",
          "delete_action": "Delete"
        }
      }
    }
  }
}