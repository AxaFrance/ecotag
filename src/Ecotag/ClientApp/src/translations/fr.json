{
  "header": {
    "title": "Un écosystème qui vise à être écologique",
    "errors": {
      "not_connected": "Non connecté"
    }
  },
  "home": {
    "title": "Accueil",
    "projects": "Projets",
    "datasets": "Datasets",
    "groups": "Groupes",
    "errors": {
      "no_roles": "Aucun rôle n'est attribué à votre profil."
    }
  }
}