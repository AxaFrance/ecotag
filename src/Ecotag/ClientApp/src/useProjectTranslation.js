import './i18n';
import {useTranslation} from "react-i18next";
import i18next from 'i18next';

const useProjectTranslation = (namespace = 'translation') => {
    const {t: translate} = useTranslation(namespace);
    return {translate};
};

export const changeProjectTranslationLanguage = (language) => {
    i18next.changeLanguage(language);
};

export default useProjectTranslation;
