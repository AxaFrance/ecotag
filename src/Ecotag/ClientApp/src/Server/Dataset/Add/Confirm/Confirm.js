import React from 'react';
import Title from '../../../../TitleBar';
import Stepper from '../../../shared/Stepper';
import Button from '@axa-fr/react-toolkit-button';
import {withRouter} from 'react-router-dom';
import './Confirm.scss';
import useProjectTranslation from "../../../../translations/useProjectTranslation";

export const Confirm = ({navBack}) => {
    const {translate} = useProjectTranslation();

    return(
        <>
            <Title title={translate('dataset.confirm.title')} goTo="/datasets" goTitle="Datasets"/>
            <Stepper activeStep="confirm" title={translate('dataset.confirm.stepper')} link="/datasets/new"/>
            <div className="af-confirm">
                <span className="glyphicon glyphicon--ring glyphicon-ok"/>
                <h2 className="af-confirm__message">{translate('dataset.confirm.confirmation_title')}</h2>
                <h4 className="af-confirm__message">{translate('dataset.confirm.confirmation_description')}</h4>
                <Button classModifier="success hasiconLeft" id="return_datatset" name="return_datatset" onClick={navBack}>
                    <span className="af-btn-text">{translate('dataset.confirm.return_button_label')}</span>
                    <span className="icons-list__item-icon glyphicon glyphicon-list-alt"/>
                </Button>
            </div>
        </>
    );
}

const ConfirmContainer = ({history}) => {
    const navBack = () => {
        history.push('/datasets');
    };
    return <Confirm navBack={navBack}/>;
};

export default withRouter(ConfirmContainer);
