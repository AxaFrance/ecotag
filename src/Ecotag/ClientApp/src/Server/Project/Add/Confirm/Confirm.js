import React from 'react';
import Title from '../../../../TitleBar';
import Stepper from '../../../shared/Stepper';
import Button from '@axa-fr/react-toolkit-button';
import {useLocation, withRouter} from 'react-router-dom';
import './Confirm.scss';
import useProjectTranslation from '../../../../translations/useProjectTranslation';

export const Confirm = ({navBack, navViewProject}) => {
    const {translate} = useProjectTranslation();

    return(
        <>
            <Title title={translate('project.confirm.title')} goTo="/projects" goTitle="Projets"/>
            <Stepper activeStep="confirm" title={translate('project.confirm.stepper')} link="/projects/new"/>
            <div className="af-confirm">
                <span className="glyphicon glyphicon--ring glyphicon-ok"/>
                <h2 className="af-confirm__message">{translate('project.confirm.new_project_added')}</h2>
                <Button classModifier="success hasiconLeft" id="view_project" name="view_project" onClick={navViewProject}>
                    <span className="af-btn-text">{translate('project.confirm.view_project_button_label')}</span>
                    <span className="icons-list__item-icon glyphicon glyphicon-list-alt"/>
                </Button>
                <Button classModifier="success hasiconLeft" id="return_datatset" name="return_datatset" onClick={navBack}>
                    <span className="af-btn-text">{translate('project.confirm.go_back_button_label')}</span>
                    <span className="icons-list__item-icon glyphicon glyphicon-list-alt"/>
                </Button>
            </div>
        </>
    );
}

export const ConfirmContainer = ({history}) => {
    const location = useLocation();
    const projectId = location.state.projectId;
    const navBack = () => {
        history.push('/projects');
    };
    const navViewProject = () => {
        history.push(`${projectId}`);
    };
    return <Confirm navBack={navBack} navViewProject={navViewProject}/>;
};

export default withRouter(ConfirmContainer);
