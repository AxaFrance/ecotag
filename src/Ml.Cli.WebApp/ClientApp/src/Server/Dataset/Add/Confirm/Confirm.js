import React from 'react';
import Title from 'TitleBar';
import Stepper from '../../../shared/Stepper';
import Button from '@axa-fr/react-toolkit-button';
import { withRouter } from 'react-router-dom';
import './Confirm.scss';

export const Confirm = ({ navBack }) => (
  <>
      <Title title="Confirmation" goTo="/datasets" goTitle="Datasets" />
    <Stepper activeStep="confirm" title="Nouveau dataset" link="/datasets/new" />
    <div className="af-confirm">
      <span className="glyphicon glyphicon--ring glyphicon-ok" />
      <h2 className="af-confirm__message">Nouveau dataset ajoutÃ© !</h2>
      <Button classModifier="success hasiconLeft" id="return_datatset" name="return_datatset" onClick={navBack}>
        <span className="af-btn-text">Retour liste datatsets</span>
        <span className="icons-list__item-icon glyphicon glyphicon-list-alt" />
      </Button>
    </div>
  </>
);

const ConfirmContainer = ({ history }) => {
  const navBack = () => {
    history.push('/datasets');
  };
  return <Confirm navBack={navBack} />;
};

export default withRouter(ConfirmContainer);
