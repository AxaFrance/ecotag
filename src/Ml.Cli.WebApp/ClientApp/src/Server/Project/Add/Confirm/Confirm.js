import React from 'react';
import Title from 'TitleBar';
import Stepper from '../../../shared/Stepper';
import Button from '@axa-fr/react-toolkit-button';
import { useLocation, withRouter } from 'react-router-dom';
import './Confirm.scss';

export const Confirm = ({ navBack, navViewProject }) => (
  <>
    <Title title="Confirmation" goTo="/projects" goTitle="Projets" />
    <Stepper activeStep="confirm" title="Nouveau projet" link="/projects/new" />
    <div className="af-confirm">
      <span className="glyphicon glyphicon--ring glyphicon-ok" />
      <h2 className="af-confirm__message">Nouveau projet ajout√© !</h2>
      <Button classModifier="success hasiconLeft" id="view_project" name="view_project" onClick={navViewProject}>
        <span className="af-btn-text">Visualiser le projet</span>
        <span className="icons-list__item-icon glyphicon glyphicon-list-alt" />
      </Button>
      <Button classModifier="success hasiconLeft" id="return_datatset" name="return_datatset" onClick={navBack}>
        <span className="af-btn-text">Retour liste des projets</span>
        <span className="icons-list__item-icon glyphicon glyphicon-list-alt" />
      </Button>
    </div>
  </>
);

export const ConfirmContainer = ({ history }) => {
  const location = useLocation();
  const project = location.state.project;
  const navBack = () => {
    history.push('/');
  };
  const navViewProject = () => {
    history.push(`${project.id}`);
  };
  return <Confirm navBack={navBack} navViewProject={navViewProject} />;
};

export default withRouter(ConfirmContainer);
