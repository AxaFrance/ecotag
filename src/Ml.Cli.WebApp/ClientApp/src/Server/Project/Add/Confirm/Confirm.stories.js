import React from 'react';
import { action } from '@storybook/addon-actions';
import { Confirm } from './Confirm';
import { BrowserRouter as Router } from 'react-router-dom';

export default { title: 'Project/Add/Confirm' };

export const withDefault = () => <Router><Confirm onClick={action('onClick')} /></Router>;
