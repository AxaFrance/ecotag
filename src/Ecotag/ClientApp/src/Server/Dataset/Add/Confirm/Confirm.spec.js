import {render} from "@testing-library/react";
import {BrowserRouter as Router} from "react-router-dom";
import React from "react";
import {Confirm} from "./Confirm";
import {changeProjectTranslationLanguage} from "../../../../translations/useProjectTranslation";

describe('Confirm', () => {
    const renderComponentAndCheckSnapshot = () => {
        const {asFragment} = render(<Router><Confirm fetch={fetch}/></Router>);
        expect(asFragment()).toMatchSnapshot();
    };

    it('Should render correctly in english', () => {
        changeProjectTranslationLanguage('en');
        renderComponentAndCheckSnapshot();
    });
    it('should render correctly in french', () => {
        changeProjectTranslationLanguage('fr');
        renderComponentAndCheckSnapshot();
    });
})