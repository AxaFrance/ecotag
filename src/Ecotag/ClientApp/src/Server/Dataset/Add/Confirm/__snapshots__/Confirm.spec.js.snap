// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Should render correctly in english 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--backhome"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/datasets"
        title="Datasets"
      >
        <i
          class="glyphicon glyphicon-arrowthin-left"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Confirmation
      </h1>
    </div>
  </div>
  <div
    class="af-steps-new af-steps-new--devis"
  >
    <ol
      class="af-steps-list"
    >
      <li
        class="past af-steps-list-step number"
        title="New dataset"
      >
        <a
          class="af-steps-list-stepLabel"
          href="/datasets/new"
        >
          <div
            class="af-steps-list-stepNumber"
          >
            1
          </div>
          <div
            class="af-steps-list-stepTitle"
          >
            New dataset
          </div>
        </a>
      </li>
      <li
        class="af-steps-list-step af-steps-list-step--success"
        title="Confirmation"
      >
        <div
          class="af-steps-list-stepLabel"
        >
          <span
            class="af-steps-list-stepTitle"
          >
            Confirm
          </span>
        </div>
      </li>
    </ol>
  </div>
  <div
    class="af-confirm"
  >
    <span
      class="glyphicon glyphicon--ring glyphicon-ok"
    />
    <h2
      class="af-confirm__message"
    >
      New dataset added !
    </h2>
    <h4
      class="af-confirm__message"
    >
      If you have imported a dataset, the process of adding files can take a while.
    </h4>
    <button
      class="btn af-btn af-btn--success af-btn--hasiconLeft"
      id="return_datatset"
      name="return_datatset"
    >
      <span
        class="af-btn-text"
      >
        Go back to datasets list
      </span>
      <span
        class="icons-list__item-icon glyphicon glyphicon-list-alt"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Confirm should render correctly in french 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--backhome"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/datasets"
        title="Datasets"
      >
        <i
          class="glyphicon glyphicon-arrowthin-left"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Confirmation
      </h1>
    </div>
  </div>
  <div
    class="af-steps-new af-steps-new--devis"
  >
    <ol
      class="af-steps-list"
    >
      <li
        class="past af-steps-list-step number"
        title="Nouveau dataset"
      >
        <a
          class="af-steps-list-stepLabel"
          href="/datasets/new"
        >
          <div
            class="af-steps-list-stepNumber"
          >
            1
          </div>
          <div
            class="af-steps-list-stepTitle"
          >
            Nouveau dataset
          </div>
        </a>
      </li>
      <li
        class="af-steps-list-step af-steps-list-step--success"
        title="Confirmation"
      >
        <div
          class="af-steps-list-stepLabel"
        >
          <span
            class="af-steps-list-stepTitle"
          >
            Confirmation
          </span>
        </div>
      </li>
    </ol>
  </div>
  <div
    class="af-confirm"
  >
    <span
      class="glyphicon glyphicon--ring glyphicon-ok"
    />
    <h2
      class="af-confirm__message"
    >
      Nouveau dataset ajouté !
    </h2>
    <h4
      class="af-confirm__message"
    >
      Si vous avez importé un dataset, le processus d'ajout des fichiers peut prendre un moment.
    </h4>
    <button
      class="btn af-btn af-btn--success af-btn--hasiconLeft"
      id="return_datatset"
      name="return_datatset"
    >
      <span
        class="af-btn-text"
      >
        Retour liste datasets
      </span>
      <span
        class="icons-list__item-icon glyphicon glyphicon-list-alt"
      />
    </button>
  </div>
</DocumentFragment>
`;
